wyrm/processing.py

```diff
@@ -1838,6 +1838,74 @@ def calculate_classwise_average(dat, classaxis=0):
     return dat.copy(data=data, axes=axes)
 
 
+def calculate_cca(dat_x, dat_y, timeaxis=-2):
+    """Calculate the Canonical Correlation Analysis (CCA) of specified data.
+
+    This method calculates the canonical correlation and the corresponding weights.
+
+    Examples
+    --------
+    Calculate the CCA for the specified multivariable signals.
+
+    >>> rho, w_x, w_y = calculate_cca(dat_x, dat_y)
+    >>> # Obtain canonical variables
+    >>> cv_x = np.dot(dat_x, w_x)
+    >>> cv_y = np.dot(dat_y, w_y)
+
+    Parameters
+    ----------
+    dat_x, dat_y : continuous Data object
+        these data should have the same length on the time axis.
+    timeaxis : int, optional
+        the index of the time axis in ``dat_x`` and ``dat_y``.
+
+    Returns
+    -------
+    rho : float
+        the canonical correlation.
+    w_x, w_y : 1d array
+        the weights corresponding to each canonical variable.
+
+    Raises
+    ------
+    AssertionError :
+        If:
+          * ``dat_x`` and ``dat_y`` is not continuous Data object
+          * the length of ``dat_x`` and ``dat_y`` is different on the ``timeaxis``
+
+    References
+    ----------
+    http://en.wikipedia.org/wiki/Canonical_correlation
+
+    """
+    assert (len(dat_x.data.shape) == len(dat_y.data.shape) == 2 and
```

Comment:
In Wyrm, the continuous Data object usually consists of a 2d array, doesn't it?
So, I think that the case of 'data.shape != 2' is included the first condition and it's unit test is
already implemented at TestCalculateCCA.test_raise_error_with_non_continuous_data.